zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/migration/MigrateCallActivityTest.java

```diff
@@ -235,4 +235,204 @@ public void shouldAllowToTerminateFlowAfterMigratingParentProcessInstance() {
         .describedAs("Expect that both the parent and child process instance terminated")
         .contains(childProcessInstanceKey, parentProcessInstanceKey);
   }
+
+  @Test
+  public void shouldMigrateChildProcessInstance() {
+    // given
+    final String processId = helper.getBpmnProcessId() + "_source";
+    final String childProcessId = helper.getBpmnProcessId() + "_child";
+    final String targetChildProcessId = helper.getBpmnProcessId() + "_target";
+    final var deployment =
+        ENGINE
+            .deployment()
+            .withXmlResource(
+                Bpmn.createExecutableProcess(processId)
+                    .startEvent("start")
+                    .callActivity("callActivity", c -> c.zeebeProcessId(childProcessId))
+                    .endEvent("end")
+                    .done())
+            .withXmlResource(
+                Bpmn.createExecutableProcess(childProcessId)
+                    .startEvent("start")
+                    .userTask("A", AbstractUserTaskBuilder::zeebeUserTask)
+                    .endEvent("end")
+                    .done())
+            .withXmlResource(
+                Bpmn.createExecutableProcess(targetChildProcessId)
+                    .startEvent("start")
+                    .userTask("B", AbstractUserTaskBuilder::zeebeUserTask)
+                    .endEvent("end")
+                    .done())
+            .deploy();
+    final long processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(processId).create();
+
+    final var childUserTask =
+        RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
+            .withParentProcessInstanceKey(processInstanceKey)
+            .withElementType(BpmnElementType.USER_TASK)
+            .withElementId("A")
+            .getFirst();
+    final var childProcessInstanceKey = childUserTask.getValue().getProcessInstanceKey();
+
+    final long targetProcessDefinitionKey =
+        extractProcessDefinitionKeyByProcessId(deployment, targetChildProcessId);
+
+    // when
+    ENGINE
+        .processInstance()
+        .withInstanceKey(childProcessInstanceKey)
+        .migration()
+        .withTargetProcessDefinitionKey(targetProcessDefinitionKey)
+        .addMappingInstruction("A", "B")
+        .migrate();
+
+    // then
+    assertThat(
+            RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_MIGRATED)
+                .withProcessInstanceKey(childProcessInstanceKey)
+                .withElementType(BpmnElementType.USER_TASK)
+                .getFirst()
+                .getValue())
+        .describedAs("Expect that process definition key is changed")
+        .hasProcessDefinitionKey(targetProcessDefinitionKey)
+        .describedAs("Expect that bpmn process id and element id changed")
+        .hasBpmnProcessId(targetChildProcessId)
+        .hasElementId("B")
+        .describedAs("Expect that version number did not change")
+        .hasVersion(1);
```

Comment:
👍 Looks good

:x: As we're migrating an entire process instance, we should also verify that the process instance itself is migrated.

Reply:
That is a very valid point, updated!